R/SingleR.visualizations.R

```diff
@@ -0,0 +1,155 @@
+#' @author Daniel Bunis, based on code by Dvir Aran.
+#' @examples
+#' ###########################################
+#' ## Mocking up some example training data ##
+#' ###########################################
+#'
+#' Ngroups <- 5
+#' Ngenes <- 1000
+#' means <- matrix(rnorm(Ngenes*Ngroups), nrow=Ngenes)
+#' means[1:900,] <- 0
+#' colnames(means) <- LETTERS[1:5]
+#'
+#' N <- 100
+#' g <- sample(LETTERS[1:5], N, replace=TRUE)
+#' train.data <- SingleCellExperiment(
+#'     list(counts=matrix(rpois(1000*N, lambda=2^means[,g]), ncol=N)),
+#'     colData=DataFrame(label=g)
+#' )
+#' rownames(train.data) <- sprintf("GENE_%s", seq_len(nrow(train.data)))
+#'
+#' ##################################################
+#' ## Mocking up some test data for classification ##
+#' ##################################################
+#'
+#' N <- 100
+#' g <- sample(LETTERS[1:5], N, replace=TRUE)
+#' sc.data <- SingleCellExperiment(
+#'     list(counts=matrix(rpois(1000*N, lambda=2^means[,g]), ncol=N)),
+#'     colData=DataFrame(label=g)
+#' )
+#' rownames(sc.data) <- sprintf("GENE_%s", seq_len(nrow(sc.data)))
+#' 
+#' #####################
+#' ## Running SingleR ##
+#' #####################
+#' 
+#' pred <- SingleR(sc.data, train.data, labels=train.data$label)
+#' table(predicted=pred$labels, truth=g)
+#' 
+#' ##### Check if cell#1 resembles reference-set cells of the same type
+#' # Identify reference cells of the same type as cell#1
+#'   #This cell is type...
+#'   pred$labels[1]
+#' ref.sameType.as1 <- grep(pred$labels[1], train.data$label)
+#' 
+#' # Compare expression of target cell to reference cells of the same type
+#' plotCellVsReference(sc.data, sc.id = 1,
+#'                     train.data, train.id = ref.sameType.as1[1])
+#' 
+#' # Compare expression of target cell to reference cells of a different type
+#' ref.diffType.as1 <- seq_along(train.data$label)[-ref.sameType.as1]
+#' plotCellVsReference(sc.data, sc.id = 1,
+#'                     train.data, train.id = ref.diffType.as1[1])
+#' 
+#' @export
+#' @importFrom ggplot2 ggplot geom_point geom_smooth theme xlab ylab ggtitle theme_classic
+#' @importFrom SummarizedExperiment assay
+#' @importClassesFrom SingleCellExperiment SingleCellExperiment
+#' @importFrom methods is
+plotCellVsReference <- function(sc.data, sc.id, train.data,train.id, assay.type.sc = 1, assay.type.train = 1) {
+    if (is(sc.data, "SingleCellExperiment")) {
+        sc.data <- assay(sc.data, assay.type.sc)
+    }
+    if (is(train.data, "SingleCellExperiment")) {
+        train.data <- assay(train.data, assay.type.train)
+    }
+    if (is(train.data, "list")) {
+        train.data <- train.data$data
+    }
+    rownames(sc.data) <- tolower(rownames(sc.data))
+    rownames(train.data) <- tolower(rownames(train.data))
+    A <- intersect(rownames(sc.data),rownames(train.data))
+    df <- data.frame(sc.data[A,sc.id],train.data[A,train.id])
+    colnames(df) <- c('x','y')
```

Comment:
Save yourself a line and just have `df <- data.frame(x=..., y=...)`.

Reply:
True!  changed.